datafusion/physical-expr/src/expressions/column.rs

```diff
@@ -176,7 +176,7 @@ impl PhysicalExpr for UnKnownColumn {
 
     /// Evaluate the expression
     fn evaluate(&self, _batch: &RecordBatch) -> Result<ColumnarValue> {
-        plan_err!("UnKnownColumn::evaluate() should not be called")
+        exec_err!("UnKnownColumn::evaluate() should not be called")
```

Comment:
If it means this line is unreachable, it should be Internal Error: if it's reached then there must be a bug in Datafusion (I remember it will print an error message to let the user file a GitHub issue for such error type)

Reply:
Makes sense 👍